### ISSUE TITLE
Set default webdriver options when call "attach" method

### ISSUE THREAD

**Issue by DudaGod** [webdriverio/webdriverio#6738]
When some of these options: `protocol`, `hostname`, `port`, `path` are not specified then default values from `detectBackend` - https://github.com/webdriverio/webdriverio/blob/72923b7bedbdd4c4e31280ba7d03450215a5998c/packages/webdriverio/src/utils/detectBackend.ts#L45 are not used because it is not call inside `attach` method.

**Environment (please complete the following information):**
 - **WebdriverIO version:** 7.1.0
 - **Mode:** Standalone mode
 - **If WDIO Testrunner, running sync/async:** async
 - **Node.js version:** 12.18.1
 - **NPM version:** 6.14.5

**Describe the bug**

Example:
```
const {remote, attach} = require('webdriverio')

let client, client2;

const opts = {
    protocol: 'http',
    hostname: 'MY_HOSTNAME',
    // port: 4444, <- not specified explicitly
    path: '/wd/hub',
    logLevel: 'trace',
    automationProtocol: 'webdriver',

    capabilities: {
        browserName: 'chrome',
        browserVersion: '85.0',
    }
};

(async function () {
    client = await remote(opts);

    client2 = await attach({
        sessionId: client.sessionId,
        ...opts
    });

    await client2.url('https://yandex.ru')

    await client.deleteSession();
})()
    .catch(async (e) => {
        console.error(e.stack)
        await client.deleteSession()
    });
```

Fails with error:
```
TypeError [ERR_INVALID_URL]: Invalid URL: http://MY_HOSTNAME:undefined/wd/hub/session/063834a0ea570f518bd2b8dcc57178fcf961b0b28030e37a52d3955238875ea9/url
    at onParseError (internal/url.js:257:9)
    at new URL (internal/url.js:333:5)
    at WebDriverRequest._createOptions (/Users/dudkevich/job/projects/dudkevich-scripts/FEI-20461.wdio_6/node_modules/webdriver/build/request.js:102:30)
    at WebDriverRequest.makeRequest (/Users/dudkevich/job/projects/dudkevich-scripts/FEI-20461.wdio_6/node_modules/webdriver/build/request.js:64:38)
    at Browser.protocolCommand (/Users/dudkevich/job/projects/dudkevich-scripts/FEI-20461.wdio_6/node_modules/webdriver/build/command.js:74:24)
    at Browser.wrapCommandFn (/Users/dudkevich/job/projects/dudkevich-scripts/FEI-20461.wdio_6/node_modules/@wdio/utils/build/shim.js:79:38)
    at Browser.url (/Users/dudkevich/job/projects/dudkevich-scripts/FEI-20461.wdio_6/node_modules/webdriverio/build/commands/browser/url.js:45:17)
    at Browser.wrapCommandFn (/Users/dudkevich/job/projects/dudkevich-scripts/FEI-20461.wdio_6/node_modules/@wdio/utils/build/shim.js:79:38)
    at /Users/dudkevich/job/projects/dudkevich-scripts/FEI-20461.wdio_6/attach-chrome.js:38:19
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

**Comment by christian-bromann** [comment]
As I mentioned in the other issue the `attach` method could be improved in many ways. Any contributions is appreiciated.

**Comment by DudaGod** [comment]
> As I mentioned in the other issue the attach method could be improved in many ways. Any contributions is appreiciated.

Yeah, I remember it, but still decided to create issue for others who will faced with the same problem. Linked issue - https://github.com/webdriverio/webdriverio/issues/6554.

**Comment by christian-bromann** [comment]
Fixed in #6759
